pkg/engine/journal.go

```diff
@@ -112,8 +113,9 @@ func (entries JournalEntries) Snap(base *deploy.Snapshot) (*deploy.Snapshot, err
 			case deploy.OpImport, deploy.OpImportReplacement:
 				resources = append(resources, e.Step.New())
 			case deploy.OpRefresh:
-				step, ok := e.Step.(*deploy.RefreshStep)
-				if ok && step.Persisted() {
+				persisted, ok := e.Step.(interface{ Persisted() bool })
+				contract.Assertf(ok, "step should implement Persisted() for refreshes")
```

Comment:
Similar to:

https://github.com/pulumi/pulumi/blob/34b1514cba665e5f88b2c4511021a1f399fd3ca2/pkg/engine/update.go#L829-L832